### PR TITLE
[Ripple] [States] Moving touch handling ownership away from the ripple view

### PR DESCRIPTION
**Pull request by yarneo**
In some cases because the Ripple handles touches to figure out when the touch is moved in and out of bounds, or if it is cancelled or ended, it will consume the touch and not allow the superview or other subviews receive the touch.

Therefore, I have moved away from the approach of having the stateful ripple view receive any ownership of the touch by setting pointInside for it to always be NO. However, to be able to have stateful ripple follow material guidelines, I have put new guidance to views adding the ripple view as a subview to pass on the standard touches to the ripple view to have it work as intended.

Documentation in our GitHub on how to integrate it correctly will be added as a follow up. Documentation in the header has been fully fleshed out and added in this PR.

The example for stateful ripple incorporates these updates and works as intended.

### CODE REVIEW

File: components/Ripple/examples/StatefulRippleExample.swift [hunk 1]
```diff
@@ -49,7 +49,13 @@ class RippleView : UIView {
     statefulRippleView.isRippleHighlighted = true
   }
 
+  override func touchesMoved(_ touches: Set<UITouch>, with event: UIEvent?) {
+    statefulRippleView.superviewTouchMoved(touches.first, event: event)
```
Comment (ianegordon) [inline components/Ripple/examples/StatefulRippleExample.swift]: Consider replacing superview with parentView to avoid subclass confusion.

Naming is hard!

Comment (ianegordon) [inline components/Ripple/examples/StatefulRippleExample.swift]: nit: Does it make sense to have superviewTouchMoved have the exact same signature as touchesMoved from a simplicity standpoint?

Essentially, the documentation would be:
If you have a ripple view you must call:
rippleView.parentViewTouchesMoved(touches, with: event) in your implementation of touchesMoved(touches, with: event)

**Review by ianegordon — DISMISSED** [review]

File: components/Ripple/examples/StatefulRippleExample.swift (hunk 1 above)
Comment (yarneo) [reply components/Ripple/examples/StatefulRippleExample.swift]: I named it superview as that is the way UIKit calls view's parents. parentView is more of a view controller notion. Do you feel strongly this should be parent?

Comment (yarneo) [reply components/Ripple/examples/StatefulRippleExample.swift]: Yes, I thought the same. However, if we want to leverage UIControl's `beginTrackingWithTouch`, it only takes in one UITouch as opposed to multiple. But happy to go either way. I am open to have it have the same signature. However, if we do that, we should probably have the same signatures for touchesEnded and touchesCancelled. In that case though, they get in a touch which isn't needed on our end, and also `cancelTracking` of UIControl doesn't even get a touch as an input. Just something to be aware of.

Comment (ianegordon) [reply components/Ripple/examples/StatefulRippleExample.swift]: Blink, blink.

Ignore that, need more caffeine!

Comment (ianegordon) [reply components/Ripple/examples/StatefulRippleExample.swift]: I would match the existing signature and pass the entire set of touches.  Seems more consistent.
I would do this for all of the touches(Begin|Moved|Ended|Cancelled) methods.

File: components/Ripple/src/MDCStatefulRippleView.m [hunk 2]
```diff
@@ -71,7 +71,6 @@ - (void)commonMDCStatefulRippleViewInit {
     _rippleColors[@(MDCRippleStateSelected | MDCRippleStateDragged)] =
         [selectionColor colorWithAlphaComponent:kDefaultRippleDraggedAlpha];
   }
-  self.userInteractionEnabled = YES;
```
Comment (ianegordon) [inline components/Ripple/src/MDCStatefulRippleView.m:74]: Shouldn't we explicitly be setting this to NO?

**Review by ianegordon — CHANGES_REQUESTED** [review]

**Review by ianegordon — APPROVED** [review]
Discussed offline.
LGTM.

**Pull request merged**